packages/connect-react/src/index.ts

```diff
@@ -45,12 +49,25 @@ export function Connect({
   location,
   options,
 }: ConnectProps): JSX.Element {
-  const [org, setOrg] = useState<Organization | null>(null)
-  const [orgError, setOrgError] = useState<Error | null>(null)
-  const [orgLoading, setOrgLoading] = useState<boolean>(false)
+  const [{ org, orgError, orgLoading }, setOrgStatus] = useState<{
+    org: Organization | null
+    orgError: Error | null
+    orgLoading: boolean
+  }>({
+    org: null,
+    orgError: null,
+    orgLoading: true,
+  })
 
   const cancelOrgLoading = useRef<Function | null>(null)
 
+  // If the connector is custom instance, it is the responsibility of the provider to memoize it.
+  const connectorUpdateValue =
+    Array.isArray(connector) || typeof connector === 'string'
+      ? JSON.stringify(connector)
+      : connector
```

Comment:
Just wondering with this, what situation might this apply?

Reply:
For example, a third party implementing their own connector (like `@aragon/connector-thegraph`), or just a fork of one of ours if they need to modify anything in it. The `connect()` function also accepts a connector instance if passed as a second parameter.